src/http/image-http.js

```diff
@@ -39,14 +41,16 @@ function getAndValidateUser(uuid) {
     });
 }
 
-function uploadImage(imageName, imageFile) {
+function uploadImage(imageName, imageFile, imageOpts) {
   logger.info('Uploading', imageName);
 
   return Promise.resolve()
     .then(() => validateMimeType(imageFile.mimetype))
     .then(() => {
       if (imageFile.mimetype === 'image/gif') {
         return imageFile.buffer;
+      } else if (imageOpts.imageText) {
+        return processImageText(imageFile.buffer, imageOpts);
       } else {
         return autoOrient(imageFile.buffer);
       }
```

Comment:
Okay seems it is done inside `processImageText`. I think this would be clearer if `processImageText` was named `processImage` and all the logic was there. I.e. autoOrient() is always done and if imageText is given, the text is added. Then all image manipulation would be capsulated inside that module